staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy_test.go

```diff
@@ -136,3 +139,182 @@ func TestPrepareForUpdate(t *testing.T) {
 		}
 	}
 }
+
+func strPtr(s string) *string { return &s }
+
+func TestStatusStrategyValidateUpdate(t *testing.T) {
+	strategy := statusStrategy{}
+	kind := schema.GroupVersionKind{
+		Version: "v1",
+		Kind:    "foo",
+		Group:   "test",
+	}
+	strategy.customResourceStrategy.validator.kind = kind
+	v := "v1"
+	structuralSchemas := map[string]*structuralschema.Structural{}
+	structuralSchema, _ := structuralschema.NewStructural(&apiextensions.JSONSchemaProps{
+		Type: "object",
+		AdditionalProperties: &apiextensions.JSONSchemaPropsOrBool{
+			Allows: true,
+		},
+	})
+	structuralSchema1, _ := structuralschema.NewStructural(&apiextensions.JSONSchemaProps{
+		Type:       "object",
+		XListType:  strPtr("set"),
+		Properties: map[string]apiextensions.JSONSchemaProps{},
+	})
+	structuralSchema.AdditionalProperties.Structural = structuralSchema1
+	structuralSchemas[v] = structuralSchema
+	strategy.structuralSchemas = structuralSchemas
+
+	ctx := context.TODO()
+	tcs := []struct {
+		name    string
+		old     *unstructured.Unstructured
+		obj     *unstructured.Unstructured
+		isValid bool
+	}{
+		{
+			name: "bothValid",
+			// changes to spec are ignored
+			old: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "test/v1",
+					"kind":       "foo",
+					"spec":       "old",
+					"numArray": []interface{}{
+						1,
+						2,
+					},
+				},
```

Comment:
I'd suggest omitting anything unnecessary (like the `spec` content) and consolidating these as much as possible to make it easy to see what is changing between testcases, e.g.

```go
{
	name:    "bothValid",
	old:     &unstructured.Unstructured{Object: map[string]interface{}{"apiVersion": "test/v1", "kind": "foo", "numArray": []interface{}{1, 2}}},
	obj:     &unstructured.Unstructured{Object: map[string]interface{}{"apiVersion": "test/v1", "kind": "foo", "numArray": []interface{}{1, 3}, "metadata": map[string]interface{}{"resourceVersion": "1"}}},
	isValid: true,
},
```

Reply:
Looks more clear.
Updated.